### ISSUE TITLE
Cyrillic node title / name

### ISSUE THREAD

**Issue by Warxcell** [sulu/sulu#5050]
| Q | A
| --- | ---
| Sulu Version | 2.0.3
| Browser Version | Not related to browser

# Actual Behavior

When creating new page with Cyrillic name error occurs:
`InvalidArgumentException: Name can not be empty in /var/www/vendor/jackalope/jackalope/src/Jackalope/Node.php:1622`

# Expected Behavior

Save the new page.

# Steps to Reproduce

Create new page in https://sulu.rocks/ with cyrillic "Тест на страница с кирилица"

# Possible Solutions

I don't know what is that name used for, but one of the possible is to use Transliterator which will always transliterate everything to LATIN. The other possible solution is to remove [this code](https://github.com/sulu/sulu/blob/master/src/Sulu/Bundle/DocumentManagerBundle/Slugifier/Urlizer.php#L600-L604). 

Full stack:
`0: "InvalidArgumentException: Name can not be empty in /var/www/vendor/jackalope/jackalope/src/Jackalope/Node.php:1622↵Stack trace:↵#0 /var/www/vendor/jackalope/jackalope/src/Jackalope/Node.php(423): Jackalope\Node->getChildPath('')↵#1 /var/www/vendor/sulu/sulu/src/Sulu/Component/DocumentManager/Subscriber/Behavior/Path/AutoNameSubscriber.php(155): Jackalope\Node->addNode('')↵#2 /var/www/vendor/symfony/event-dispatcher/EventDispatcher.php(304): Sulu\Component\DocumentManager\Subscriber\Behavior\Path\AutoNameSubscriber->handlePersist(Object(Sulu\Component\DocumentManager\Event\PersistEvent), 'sulu_document_m...', Object(Symfony\Component\EventDispatcher\EventDispatcher))↵#3 /var/www/vendor/symfony/event-dispatcher/EventDispatcher.php(264): Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure}(Object(Sulu\Component\DocumentManager\Event\PersistEvent), 'sulu_document_m...', Object(Symfony\Component\EventDispatcher\EventDispatcher))↵#4 /var/www/vendor/symfony/event-dispatcher/EventDispatcher.php(239): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'sulu_document_m...', Object(Sulu\Component\DocumentManager\Event\PersistEvent))↵#5 /var/www/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners(Array, 'sulu_document_m...', Object(Sulu\Component\DocumentManager\Event\PersistEvent))↵#6 /var/www/vendor/sulu/sulu/src/Sulu/Component/DocumentManager/DocumentManager.php(69): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Sulu\Component\DocumentManager\Event\PersistEvent), 'sulu_document_m...')↵#7 /var/www/vendor/sulu/sulu/src/Sulu/Bundle/PageBundle/Controller/PageController.php(587): Sulu\Component\DocumentManager\DocumentManager->persist(Object(Sulu\Bundle\PageBundle\Document\PageDocument), 'bg', Array)↵#8 /var/www/vendor/sulu/sulu/src/Sulu/Bundle/PageBundle/Controller/PageController.php(309): Sulu\Bundle\PageBundle\Controller\PageController->persistDocument(Object(Symfony\Component\HttpFoundation\Request), 'Sulu\\Bundle\\Pag...', Object(Sulu\Bundle\PageBundle\Document\PageDocument), 'bg')↵#9 /var/www/vendor/symfony/http-kernel/HttpKernel.php(146): Sulu\Bundle\PageBundle\Controller\PageController->postAction(Object(Symfony\Component\HttpFoundation\Request))↵#10 /var/www/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)↵#11 /var/www/vendor/symfony/http-kernel/Kernel.php(201): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)↵#12 /var/www/public/index.php(63): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))↵#13 {main}"`

**Comment by danrot** [comment]
This is actually a duplicate of #3367, therefore I am going to close this issue.

**Comment by alexander-schranz** [comment]
The title is generated by the [PathCleanup](https://github.com/sulu/sulu/blob/2.0.3/src/Sulu/Component/PHPCR/PathCleanup.php) service which I think is also used to generate the urls (not sure). And is the case currently non latin character are stripped from path. Think to support non latin urls both the path cleanup and the transliterator need to be extended. We should keep an eye on BC here as I did also saw projects using the path cleanup  with `new PathCleanup()` in some entities. 

Basically not sure if we should merge make an own translitator interface or something and have a single service generating urls or paths out of a string as I think there are currently 2 services needed to extend this functionality.
